crates/svm-runtime/src/import.rs

```diff
@@ -1,52 +1,173 @@
+use std::convert::{TryFrom, TryInto};
 use std::ffi::c_void;
-use std::ptr;
+use std::rc::Rc;
 
 use crate::Context;
-use svm_types::WasmType;
 
-use wasmer::{Export, FunctionType, Type as WasmerType};
-use wasmer_vm::{ExportFunction, VMContext, VMFunctionKind};
+use wasmer::{Export, Exportable, Function, FunctionType, RuntimeError, Store, Type, Val};
 
-#[derive(Debug, PartialEq, Clone)]
-pub struct Import {
-    pub name: String,
+use svm_ffi::{svm_byte_array, svm_env_t, svm_func_callback_t, svm_trap_t};
+use svm_types::{WasmType, WasmValue};
 
-    pub params: Vec<WasmType>,
+#[derive(Debug, Clone)]
+pub struct ExternImport {
+    name: String,
 
-    pub returns: Vec<WasmType>,
+    namespace: String,
 
-    pub func_ptr: *const c_void,
+    params: Vec<WasmType>,
+
+    returns: Rc<Vec<WasmType>>,
```

Comment:
@moshababo 
I've thought about how to simplify the code of the `results` allocations and still gain performance.

So I've picked using `Rc` to share the `returns` types with the `inner_callback` using shallow clone.
(the runtime overhead is minimal and should be much less expansive than cloning the `Vec`.

Reply:
Looks good.